src/checks/spatial/buddy_check.rs

```diff
@@ -152,6 +154,34 @@ pub fn buddy_check(
     Ok(flags)
 }
 
+/// Apply [`buddy_check`] to a whole [`DataCache`]
+pub fn buddy_check_cache(
+    cache: &DataCache,
+    args: &BuddyCheckArgs,
+    // TODO: should we allow different obs_to_check for each timeslice?
+    obs_to_check: Option<&[bool]>,
+) -> Result<Vec<(String, Vec<Flag>)>, Error> {
+    let series_len = cache.data[0].1.len();
+
+    let mut result_vec: Vec<(String, Vec<Flag>)> = cache
+        .data
+        .iter()
+        .map(|ts| (ts.0.clone(), Vec::with_capacity(series_len)))
+        .collect();
+
+    for i in (cache.num_leading_points as usize)..(series_len - cache.num_trailing_points as usize)
+    {
+        let timeslice: Vec<Option<f32>> = cache.data.iter().map(|v| v.1[i]).collect();
+        let spatial_result = buddy_check(&timeslice, &cache.rtree, args, obs_to_check)?;
+
+        for i in 0..spatial_result.len() {
+            result_vec[i].1.push(spatial_result[i]);
+        }
+    }
+
+    Ok(result_vec)
```

Comment:
```suggestion
    let result_vec = ((cache.num_leading_points as usize)
        ..(series_len - cache.num_trailing_points as usize))
        .map(|i| {
            let timeslice: Vec<Option<f32>> = cache.data.iter().map(|v| v.1[i]).collect();
            Ok((
                cache.data[0].0.clone(),
                buddy_check(&timeslice, &cache.rtree, args, obs_to_check)?,
            ))
        })
        .collect();

    result_vec
```
But I'm not sure if the `String`s are right the way they are (in either version)

Reply:
after conversation, wontfix